cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js

```diff
@@ -1,212 +1,256 @@
 /*
 Copyright SecureKey Technologies Inc. All Rights Reserved.
+Copyright Avast Software. All Rights Reserved.
 
 SPDX-License-Identifier: Apache-2.0
 */
 
+import axios from "axios";
+import { encode, decode } from "js-base64";
+
 /**
- * PrepareIssuanceRequest fetches the issuer metadata and prepares an issuance transaction,
- * returning to the caller for an opportunity to consent.
+ * OpenID4CI module provides APIs for wallets to receive verifiable credentials through OIDC for Credential Issuance.
  *
- * @param {string} issuer_uri - uri of issuer server: TODO decide if this is specifically the metadata endpoint.
- * @param {Object} options
- * @param {string} options.op_state - (optional) op_state parameter for issuer-initiated transactions.
+ * @module OpenID4CI
  *
- * @returns {Promise<Object>} - If successful, returns the issuer metadata and a continuation handler function.
- *   Issuer metadata may be used to display a UI showing the fields the issuer will include in each potential credential,
- *   for user selection/consent. Call the continuation function if wallet user consents to request a credential.
  */
-export async function PrepareIssuanceRequest(
-  issuer_uri = '',
-  {
-    op_state = '',
-  }
-) {
-  // stub
-  const fetchIssuerMetadata = async function (uri) {
-    // data returned by issuer server
-    return {
-      issuer: uri,
-      authorization_endpoint: uri + '/auth',
-      token_endpoint: uri + '/token',
-      pushed_authorization_request_endpoint: uri + '/par',
-      require_pushed_authorization_requests: true,
-    };
-  };
 
-  // TODO: Will issuer metadata be at a .well-known location, or should we expect the issuer_uri parameter to be the
-  //  complete uri for the metadata endpoint
-  const issuerMetadata = await fetchIssuerMetadata(issuer_uri);
+/**
+ * authorize is used by a wallet to authorize an issuer's OIDC verifiable-credential Issuance Request.
+ *
+ * @param{Object} req - the Issuance Request from an OIDC Issuer that the wallet intends to authorize.
+ * @param{string} userPin - Optional. A 2FA PIN provided by the Issuer to the wallet through a separate channel
+ * from the request, for pre-authorized issuance flows.
+ * @param{Object} clientConfig -
+ * @param{string} clientConfig.walletCallbackURI - wallet-hosted URI for callback redirect from issuer, after user authorizes issuer.
+ * @param{string} clientConfig.userDID - the DID of the wallet user. The requested credential will be bound to this DID.
+ * @param{string} clientConfig.clientID - the wallet app instance's OIDC client ID.
+ * @param{function} jwtSigner
+ *
+ */
+export async function authorize(
+  req,
+  userPin = '',
+  clientConfig,
+  jwtSigner = async function(iss, aud, iat, c_nonce){return ''},
```

Comment:
just curious,

Should default signer throw error instead?

Reply:
Done.